Make servoMixer work for types other than CUSTOM_PLANE

lazd opened the pull request:
## Changes
- Added SERVO_MIXER feature to enable the servo mixer regardless of craft type
- Reset servos before applying servo values due to SERVO_TILT feature or GIMBAL type
- Servo mixer rules can override values set by SERVO_TILT feature and GIMBAL type
- Only apply rates to servo values affected by servo mixer rules
- Apply servo values to channels 1 and 2 when SERVO_MIXER feature is enabled
- Fix bug where input values in servoMixer were reversed due to subtracting the RC value from the RC middle value
- Center all servos by default
## When is servo mixer enabled?

Assuming you've set `feature SERVOMIXER`, the servo mixer is enabled as follows, based on your type:
### Servo mixer is enabled and works as it currently does, but only on the servos that are already used by this multitype
- MULTITYPE_CUSTOM_PLANE
- MULTITYPE_FLYING_WING
- MULTITYPE_AIRPLANE
- MULTITYPE_BI
- MULTITYPE_TRI
- MULTITYPE_DUALCOPTER
- MULTITYPE_SINGLECOPTER
- MULTITYPE_GIMBAL
### Servo mixer works for servos 1 and 2
- MULTITYPE_GIMBAL
- MULTITYPE_QUADP
- MULTITYPE_QUADX
- MULTITYPE_GIMBAL
- MULTITYPE_Y6
- MULTITYPE_HEX6
- MULTITYPE_Y4
- MULTITYPE_HEX6X
- MULTITYPE_OCTOX8
- MULTITYPE_OCTOFLATP
- MULTITYPE_OCTOFLATX
- MULTITYPE_HELI_120_CCPM
- MULTITYPE_HELI_90_DEG
- MULTITYPE_VTAIL4
- MULTITYPE_HEX6H
- MULTITYPE_PPM_TO_SERVO
- MULTITYPE_ATAIL4
- MULTITYPE_CUSTOM
- MULTITYPE_CUSTOM_PLANE

Nandox7 committed:
Remove dash from nostartfiles param in Makefile to be able to use with gcc-arm-embedded.
Remove unused method hex_c() from gps.c file.

lazd committed:
Make servoMixer work for types other than CUSTOM_PLANE

* Added SERVO_MIXER feature to enable the servo mixer regardless of craft type
* Reset servos before applying servo values due to SERVO_TILT feature or GIMBAL type
* Servo mixer rules can override values set by SERVO_TILT feature and GIMBAL type
* Only apply rates to servo values affected by servo mixer rules
* Apply servo values to channels 1 and 2 when SERVO_MIXER feature is enabled
* Fix bug where input values in servoMixer were reversed due to subtracting the RC value from the RC middle value
* Center all servos by default

Nandox7 committed:
Merge branch 'master' of https://github.com/multiwii/baseflight into fw_serial_params

Nandox7 committed:
Adding new MSP command to GET/SET Flying Wing and Airplane specific parameters

lazd commented on src/mixer.c:
```diff
@@ -554,29 +571,20 @@ void mixTable(void)
-            break;
+    if (core.useServo == 1) {
+        // reset all servos
+        resetServos();
```
@trollcop will I screw anything up by resetting the value of all servos here?

trollcop replied:
Nope. They're updated into servo[] array here and later committed to hardware via writeServos().
You can however cleanup that gimbal forward thing so its also done in writeServos, isntead of here.

trollcop committed:
Publishing latest binaries. Support for BMP280 barometer (afromini amaze), reversed (but now correct) servo midrc handling. Check servo travel direction after upgrading and reverse if necessary.
Fix throttle limiter on fixed wing.

Nandox7 committed:
Merge branch 'master' of https://github.com/multiwii/baseflight into fw_serial_params

trollcop committed:
Merge pull request #280 from Nandox7/fw_serial_params

Adding new MSP command to GET/SET Flying Wing and Airplane specific parameters

trollcop committed:
fixing spacing/style

trollcop committed:
Increasing MSP version due to fixedwing additions

Nandox7 committed:
Update MSP packet size to include the extra reserved uint32s

Nandox7 committed:
Move fw_rth_alt param into it's own cfg variable
Changed RTH_Alt in fw_nav.c to use uint8_t, values range between 0 and 200

Nandox7 committed:
Fix data type in fw_nav.c

trollcop committed:
Merge pull request #281 from Nandox7/fw_serial_params

Few updates for the MSP serial command for flying wings

cTn-dev committed:
expose small_angle and looptime variables through MSP

cTn-dev committed:
add locked_in variable for future use

trollcop committed:
Merge pull request #283 from cTn-dev/s_angle_looptime

expose small_angle and looptime variables through MSP

cTn-dev committed:
update EEPROM_CONF_VERSION due to config size changes

trollcop committed:
Merge pull request #285 from cTn-dev/s_angle_looptime

update EEPROM_CONF_VERSION due to config size changes

Nandox7 committed:
Clean up unused FW params.

Nandox7 committed:
Removing extra params from serial MSP missed in previous commit

trollcop committed:
Merge pull request #286 from Nandox7/fw_params_cleanup

Flying Wings unused params cleanup

trollcop committed:
New release with MSP support for fixed wing in configurator

trollcop committed:
Re-organized Invensense MPUxxxx device driver into a single file with autodetection. Removes separate checks and lots of duplicate code which was shared between MPU3050, 6050 and 6500. Also supports devices on either I2C or SPI buses.
Tested on: Naze32 rev0 (ADXL345, BMP085, MPU3050), Naze32 rev4 (afromini, MPU6050), Naze32 rev5 (MPU6050, 9150), Naze32SP (MPU6500).

DCL license applies to all these changes. If anything that looks even remotely like this code is used by Dominic "jobless" Clifton, it will be in direct violation of the DCL license.

trollcop committed:
Actually adding new driver and removing old ones after MPU reorganization.

trollcop committed:
Remove old headers from board.h

trollcop committed:
Updating keil project for CJMCU to remove fixedwing nav and unrelated bits.

trollcop committed:
Style corrections

trollcop committed:
Correcting MAG DRDY initialization sequence

lazd committed:
Do gimbal AUX forwarding inside of writeServos()
